src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -872,11 +872,13 @@ const WorkflowView = _.flow(
           errors,
           onBrowse: name => this.setState({ variableSelected: name }),
           onChange: (name, v) => this.setState(_.set(['modifiedConfig', key, name], v)),
-          onSetDefaults: () => {
-            this.setState(_.set(['modifiedConfig', 'outputs'], _.fromPairs(_.map(({ name }) => {
-              return [name, `this.${_.last(name.split('.'))}`]
-            }, modifiedInputsOutputs.outputs))))
-          },
+          onSetDefaults: () => this.setState(oldState => {
```

Comment:
I think the old form is technically okay in this case, but I understand that lgtm has no way of knowing that.

Reply:
Its issue is that `modifiedInputsOutputs` is being read from old state, not the fresh callback state.

Reply:
Yes, and I have a loose grasp on why that can sometimes be a problem. What I don't understand is what could actually go wrong in this particular case. I'm curious to understand this better. Let's chat in person.